### PR TITLE
Added BigQuery capability to BigQuery SQL Engine

### PR DESCRIPTION
**Pull request by fernst**
Depends on https://github.com/cdapio/cdap/pull/14178

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/relational/SQLExpressionFactory.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class SQLExpressionFactory implements ExpressionFactory<String> {
    */
   @Override
   public ExpressionFactoryType<String> getType() {
-    return StringExpressionFactoryType.SQL;
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/relational/SQLExpressionFactory.java]: Please don't change expression factory type. Use capabilities

File: src/main/java/io/cdap/plugin/gcp/bigquery/relational/SQLExpressionFactory.java [hunk 2]
```diff
@@ -51,6 +52,7 @@ public Expression compile(String expression) {
   @Override
   public Set<Capability> getCapabilities() {
     return new HashSet<>(Arrays.asList(StringExpressionFactoryType.SQL,
+                                       StandardSQLCapabilities.BIGQUERY,
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/relational/SQLExpressionFactory.java]: nit: given this grows up, it makes sense to make it a constant

Comment (tivv) [reply src/main/java/io/cdap/plugin/gcp/bigquery/relational/SQLExpressionFactory.java]: Please update javadoc, it does not reflect the implementation anymore

Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/bigquery/relational/SQLExpressionFactory.java]: Updated

**Review by tivv — APPROVED** [review]

**Pull request merged**